modules/json_data_manager.rb

```diff
@@ -32,4 +32,32 @@ def load_from_json(file_path)
     # * return to be iterated over for the UI.
     parsed_data
   end
+
+  def read_json_file(file_path)
+    load_from_json(file_path)
+  end
+
+  def get_labels_and_titles(files)
+    labels = Set.new
+    titles_by_label = Hash.new { |hash, key| hash[key] = [] }
+
+    files.each do |file|
+      data = read_json_file(file)
+
+      data.each do |item|
+        labels << item['label']
+        titles_by_label[item['label']] << item['title']
+      end
+    end
+    [labels.to_a, titles_by_label]
+  end
+
+  def display_labels_and_titles(labels, titles_by_label)
+    labels.each do |label|
+      titles = titles_by_label[label]
+      puts "Label: #{label}"
+      puts "Titles: #{titles.join(', ')}" unless titles.empty?
+      puts "\n"
+    end
+  end
```

Comment:
I really like this implementation @demesameneshoa! 💪🏼 I think it could potentially come in handy for getting and displaying all of the associated class `genre`s, `author`s, `label`s.

I will kindly suggest removing the titles from the equation since it is not part of the requirements, also this change will make the algorithm more flexible and reusable for the mentioned above! this way i.e: it can take the type of associated class we want to get and display! 🎯

This way we can avoid having duplicated code to also get and display the remaining associated classes authors and genres.

Reply:
@ITurres Thanks for pointing this one out, the changes i made now should improve our code. I have refactored the code to get data by attribute to get author, label , and genre by passing an argument